KerbalVR_Mod/KerbalVR/InteractionSystem/KerbalVR_Hand.cs

```diff
@@ -168,53 +180,87 @@ public void Initialize(SteamVR_Input_Sources handType, Hand otherHand, bool isIV
 			SteamVR_Action_Boolean actionGrab = SteamVR_Input.GetBooleanAction("default", "GrabGrip");
 			actionGrab[handType].onChange += OnChangeGrab;
 
-			// set up skeleton helper
-			if (profile.useSkeletonHelper)
-			{
-				KerbalSkeletonHelper kerbalSkeletonHelper = handObject.AddComponent<KerbalSkeletonHelper>();
-				kerbalSkeletonHelper.retargetableSetting = profile.retargetableSetting;
-				kerbalSkeletonHelper.sourceSkeletonRoot = handSkeleton.skeletonRoot;
-				kerbalSkeletonHelper.destinationSkeletonRoot = handObject.transform.Find(profile.retargetableSetting.destinationSkeletonRootPath);
-			}
+			#endregion
+
+			#region Setup hand model
+
+			IVAProfile = handSetting.GetProfile(true);
+			SetupModel(IVAProfile, ref IVAObject);
+
+			EVAProfile = handSetting.GetProfile(false);
+			SetupModel(EVAProfile, ref EVAObject);
+
+			#endregion
+
+			#region Setup other stuffs
 
 			// set up ladder
 			ladder = gameObject.AddComponent<VRLadder>();
 
 			// set up UI hand  
 			UIHand = gameObject.AddComponent<VRUIHand>();
 
-			#region Setup Colliders
+			#endregion
+
+			#region Setup Collider
 
 			// add fingertip collider for "mouse clicks"
-			fingertipTransform = handObject.transform.Find(profile.indexTipTransformPath);
-			fingertipTransform.localPosition += profile.fingertipOffset;
+			fingertipTransform = new GameObject("KVR_FingertipCollider").transform;
+			fingertipTransform.SetParent(handObject.transform);
 			fingertipCollider = fingertipTransform.gameObject.AddComponent<FingertipCollider>();
 			fingertipCollider.Initialize(this);
 
 			// this has to be after the fingertip collider is initialized and before the hand collider is initialized (for ladder setup)
 			Detach(true);
 
 			// create a child object for the colider so that it can be on a different layer
-			var palmTransform = new GameObject("handTransform").transform;
-			palmTransform.SetParent(handObject.transform.Find(profile.gripTransformPath), false);
-			palmTransform.rotation = handObject.transform.rotation;
-			palmTransform.localPosition = GripOffset;
-			handCollider = palmTransform.gameObject.AddComponent<HandCollider>();
-			handCollider.Initialize(this);
+			palmTransform = new GameObject("KVR_PalmCollider").transform;
+			palmTransform.SetParent(handObject.transform);
+			palmCollider = palmTransform.gameObject.AddComponent<HandCollider>();
+			palmCollider.Initialize(this, ladder);
 
 			// thumb is used to calculate position of pinch collider
-			thumbTransform = handObject.transform.Find(profile.thumbTipTransformPath);
+			thumbTransform = new GameObject("KVR_Thumb").transform;
+			thumbTransform.SetParent(handObject.transform);
 
 			// set up pinch behavior
-			pinchTransform = new GameObject("pinchTransform").transform;
-			pinchTransform.SetParent(handObject.transform, false);
+			pinchTransform = new GameObject("KVR_PinchCollider").transform;
+			pinchTransform.SetParent(handObject.transform);
```

Comment:
this one actually makes sense because we *have* to dynamically keep updating its position relative to two fingers